File: app/src/main/java/com/example/lucknowcityguide/WeatherService.java
package com.example.lucknowcityguide;

import com.example.lucknowcityguide.model_utils.Example;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface WeatherService {
    @GET("2.5/onecall")
    Call<Example> getWeather(@Query("lat") double lat, @Query("lon") double lon, @Query(value="exclude",encoded =true) String exclude, @Query("appid") String apiKey);
}

File: app/src/main/java/com/example/lucknowcityguide/WeatherService.kt
package com.example.lucknowcityguide

import com.example.lucknowcityguide.model_utils.Example
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface WeatherService {
    @GET("2.5/onecall")
    fun getWeather(
        @Query("lat") lat: Double,
        @Query("lon") lon: Double,
        @Query(value = "exclude", encoded = true) exclude: String?,
        @Query("appid") apiKey: String?
    ): Call<Example?>?
}